packages/tools/src/tools/SculptorTool.ts

```diff
@@ -0,0 +1,590 @@
+    indicesToInsertAfter: Array<number>,
+    maxSpacing: number
+  ): void {
+    const { element } = this.sculptData;
+    const enabledElement = getEnabledElement(element);
+    const { viewport } = enabledElement;
+    const nextHandleIndex = contourIndex(i + 1, points.length);
+
+    const currentCanvasPoint = viewport.worldToCanvas(points[i]);
+    const nextCanvasPoint = viewport.worldToCanvas(points[nextHandleIndex]);
+
+    const distanceToNextHandle = point.distanceToPoint(
+      currentCanvasPoint,
+      nextCanvasPoint
+    );
+
+    if (distanceToNextHandle > maxSpacing) {
+      indicesToInsertAfter.push(i);
+    }
+  }
+
+  /**
+   * Updates cursor size
+   *
+   * @param evt - The event
+   */
+  private updateCursor(evt: EventTypes.InteractionEventType): void {
+    const eventData = evt.detail;
+    const element = eventData.element;
+
+    const enabledElement = getEnabledElement(element);
+    const { renderingEngine, viewport } = enabledElement;
+
+    this.commonData.viewportIdsToRender = [viewport.id];
+
+    const annotations = this.filterSculptableAnnotationsForElement(element);
+
+    if (!annotations?.length) {
+      return;
+    }
+
+    const activeAnnotation = annotations.find(
+      (annotation) =>
+        annotation.annotationUID === this.commonData.activeAnnotationUID
+    );
+
+    this.commonData.canvasLocation = eventData.currentPoints.canvas;
+
+    if (this.isActive) {
+      activeAnnotation.highlighted = true;
+    } else {
+      const cursorShape = this.registeredShapes.get(this.selectedShape);
+      const canvasCoords = eventData.currentPoints.canvas;
+      cursorShape.UpdateToolsize(canvasCoords, viewport, activeAnnotation);
+    }
+
+    triggerAnnotationRenderForViewportIds(
+      renderingEngine,
+      this.commonData.viewportIdsToRender
+    );
+  }
+
+  /**
+   * Returns sculptable freehand ROI annotations
+   *
+   * @param element - The viewport element
+   */
+  private filterSculptableAnnotationsForElement(
+    element: HTMLDivElement
+  ): Annotations {
+    const config = this.configuration;
+    const enabledElement = getEnabledElement(element);
+    const { renderingEngineId, viewportId } = enabledElement;
+    let sculptableAnnotations = [];
+
+    const toolGroup = ToolGroupManager.getToolGroupForViewport(
+      viewportId,
+      renderingEngineId
+    );
+
+    const toolInstance = toolGroup.getToolInstance(config.referencedToolName);
+
+    config.referencedToolNames.forEach((referencedToolName: string) => {
+      const annotations = getAnnotations(referencedToolName, element);
+      if (annotations) {
+        sculptableAnnotations = [...sculptableAnnotations, ...annotations];
+      }
+    });
+
+    sculptableAnnotations =
+      toolInstance.filterInteractableAnnotationsForElement(
+        element,
+        sculptableAnnotations
+      );
+
+    return sculptableAnnotations;
+  }
+
+  private configureToolSize(evt: EventTypes.InteractionEventType): void {
```

Comment:
Can you add a comment saying this calls the shape specific configure tool size?  I had to read the code to figure out why this one was different from the earlier tool size method